sdk/azidentity/interactive_browser_credential.go

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azidentity
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/pkg/browser"
+)
+
+// InteractiveBrowserCredentialOptions can be used when providing additional credential information, such as a client secret.
+// Also use these options to modify the default pipeline behavior through the TokenCredentialOptions.
+type InteractiveBrowserCredentialOptions struct {
+	// The Azure Active Directory tenant (directory) ID of the service principal.
+	TenantID *string
+	// The client (application) ID of the service principal.
+	ClientID *string
+	// The client secret that was generated for the App Registration used to authenticate the client. Only applies for web apps.
+	ClientSecret *string
+	// The redirect URI used to request the authorization code. Must be the same URI that is configured for the App Registration.
+	RedirectURI *string
+	// Options allows configuring the management of the requests sent to Azure Active Directory.
+	Options *TokenCredentialOptions
+}
+
+// InteractiveBrowserCredential enables authentication to Azure Active Directory using an interactive browser to log in.
+type InteractiveBrowserCredential struct {
+	client *aadIdentityClient
+	// Gets the Azure Active Directory tenant (directory) ID of the service principal.
+	tenantID string
+	// Gets the client (application) ID of the service principal.
+	clientID string
+	// Gets the client secret that was generated for the App Registration used to authenticate the client.
+	clientSecret string
+	// The redirect URI used to request the authorization code. Must be the same URI that is configured for the App Registration.
+	redirectURI string
+}
+
+// NewInteractiveBrowserCredential constructs a new InteractiveBrowserCredential with the details needed to authenticate against Azure Active Directory through an interactive browser window.
+// tenantID: The Azure Active Directory tenant (directory) ID of the service principal.
+// clientID: The client (application) ID of the service principal.
+// clientSecret: Gets the client secret that was generated for the App Registration used to authenticate the client.
+// options: allow to configure the management of the requests sent to Azure Active Directory.
+func NewInteractiveBrowserCredential(options *InteractiveBrowserCredentialOptions) (*InteractiveBrowserCredential, error) {
+	var credentialOptions *TokenCredentialOptions
+	tenantID := "organizations"
+	clientID := developerSignOnClientID
+	redirectURI := ""
```

Comment:
This, and the above defaults, should all go in a `DefaultInteractiveBrowserCredentialOptions()` method.

Reply:
Done